diff --git a/cinder/tests/unit/volume/drivers/test_quobyte.py b/cinder/tests/unit/volume/drivers/test_quobyte.py
index fe422fc23..c7308b079 100644
--- a/cinder/tests/unit/volume/drivers/test_quobyte.py
+++ b/cinder/tests/unit/volume/drivers/test_quobyte.py
@@ -73,6 +73,20 @@ class QuobyteDriverTestCase(test.TestCase):
     SNAP_UUID_2 = 'bebedede-bebe-dede-bebe-dedebebedede'
     CACHE_NAME = quobyte.QuobyteDriver.QUOBYTE_VOLUME_SNAP_CACHE_DIR_NAME
 
+    QEMU_IMG_INFO_TEMPLATE_SNAP = """{
+    "filename": "%s",
+    "format": "raw",
+    "virtual-size": 1073741824,
+    "actual-size": 173000,
+    "backing-filename": "%s"
+}"""
+    QEMU_IMG_INFO_TEMPLATE_SRC = """{
+    "filename": "%s",
+    "format": "raw",
+    "virtual-size": 1073741824,
+    "actual-size": 173000
+}"""
+
     def _get_fake_snapshot(self, src_volume):
         snapshot = fake_snapshot.fake_snapshot_obj(
             self.context,
@@ -990,36 +1004,46 @@ class QuobyteDriverTestCase(test.TestCase):
 
         size = dest_volume['size']
 
-        qemu_img_output = """{
-    "filename": "%s",
-    "format": "raw",
-    "virtual-size": 1073741824,
-    "actual-size": 173000,
-    "backing-filename": "%s"
-}""" % (snap_file, src_volume['name'])
-        img_info = imageutils.QemuImgInfo(qemu_img_output, format='json')
+        qemu_img_output_snap = self.QEMU_IMG_INFO_TEMPLATE_SNAP % (
+            snap_file, src_volume['name'])
+        img_info_snap = imageutils.QemuImgInfo(
+            qemu_img_output_snap, format='json')
+
+        qemu_img_output_bck = self.QEMU_IMG_INFO_TEMPLATE_SRC % (
+            src_volume['name'])
+        img_info_src = imageutils.QemuImgInfo(
+            qemu_img_output_bck, format='json')
 
         # mocking and testing starts here
         mock_convert = self.mock_object(image_utils, 'convert_image')
         drv._read_info_file = mock.Mock(return_value=
                                         {'active': snap_file,
                                          snapshot['id']: snap_file})
-        self.mock_object(image_utils, 'qemu_img_info', return_value=img_info)
+        self.mock_object(image_utils, 'qemu_img_info',
+                         side_effect=[img_info_snap, img_info_src])
         drv._set_rw_permissions = mock.Mock()
 
         drv._copy_volume_from_snapshot(snapshot, dest_volume, size)
 
         drv._read_info_file.assert_called_once_with(info_path)
-        image_utils.qemu_img_info.assert_called_once_with(
+        snap_info_call = mock.call(
             snap_path,
             force_share=True,
             run_as_root=False,
             allow_qcow2_backing_file=True)
+        src_info_call = mock.call(
+            src_vol_path,
+            force_share=True,
+            run_as_root=False,
+            allow_qcow2_backing_file=True)
+        image_utils.qemu_img_info.assert_has_calls([snap_info_call,
+                                                    src_info_call])
         (mock_convert.
          assert_called_once_with(src_vol_path,
                                  dest_vol_path,
                                  'raw',
-                                 run_as_root=self._driver._execute_as_root))
+                                 run_as_root=self._driver._execute_as_root,
+                                 data=img_info_src))
         drv._set_rw_permissions.assert_called_once_with(dest_vol_path)
 
     @mock.patch.object(quobyte.QuobyteDriver, "_fallocate_file")
@@ -1036,6 +1060,7 @@ class QuobyteDriverTestCase(test.TestCase):
 
         vol_dir = os.path.join(self.TEST_MNT_POINT_BASE,
                                drv._get_hash_str(self.TEST_QUOBYTE_VOLUME))
+        src_vol_path = os.path.join(vol_dir, src_volume['name'])
         dest_vol_path = os.path.join(vol_dir, dest_volume['name'])
         info_path = os.path.join(vol_dir, src_volume['name']) + '.info'
 
@@ -1049,35 +1074,44 @@ class QuobyteDriverTestCase(test.TestCase):
 
         size = dest_volume['size']
 
-        qemu_img_output = """{
-    "filename": "%s",
-    "format": "raw",
-    "virtual-size": 1073741824,
-    "actual-size": 173000,
-    "backing-filename": "%s"
-}""" % (snap_file, src_volume['name'])
-        img_info = imageutils.QemuImgInfo(qemu_img_output, format='json')
+        qemu_img_output_snap = self.QEMU_IMG_INFO_TEMPLATE_SNAP % (
+            snap_file, src_volume['name'])
+        img_info_snap = imageutils.QemuImgInfo(qemu_img_output_snap,
+                                               format='json')
+
+        qemu_img_output_bck = self.QEMU_IMG_INFO_TEMPLATE_SRC % (
+            src_volume['name'])
+        img_info_src = imageutils.QemuImgInfo(
+            qemu_img_output_bck, format='json')
 
         # mocking and testing starts here
         mock_convert = self.mock_object(image_utils, 'convert_image')
         drv._read_info_file = mock.Mock(return_value=
                                         {'active': snap_file,
                                          snapshot['id']: snap_file})
-        self.mock_object(image_utils, 'qemu_img_info', return_value=img_info)
+        self.mock_object(image_utils, 'qemu_img_info',
+                         side_effect=[img_info_snap, img_info_src])
         drv._set_rw_permissions = mock.Mock()
         shutil.copyfile = mock.Mock()
-
-        drv._copy_volume_from_snapshot(snapshot, dest_volume, size)
-
-        drv._read_info_file.assert_called_once_with(info_path)
-        image_utils.qemu_img_info.assert_called_once_with(
+        snap_info_call = mock.call(
             snap_path,
             force_share=True,
             run_as_root=False,
             allow_qcow2_backing_file=True)
+        src_info_call = mock.call(
+            src_vol_path,
+            force_share=True,
+            run_as_root=False,
+            allow_qcow2_backing_file=True)
+
+        drv._copy_volume_from_snapshot(snapshot, dest_volume, size)
+
+        drv._read_info_file.assert_called_once_with(info_path)
         self.assertFalse(mock_convert.called,
                          ("_convert_image was called but should not have been")
                          )
+        image_utils.qemu_img_info.assert_has_calls([snap_info_call,
+                                                    src_info_call])
         os_ac_mock.assert_called_once_with(
             drv._local_volume_from_snap_cache_path(snapshot), os.F_OK)
         qb_falloc_mock.assert_called_once_with(dest_vol_path, size)
@@ -1113,39 +1147,49 @@ class QuobyteDriverTestCase(test.TestCase):
 
         size = dest_volume['size']
 
-        qemu_img_output = """{
-    "filename": "%s",
-    "format": "raw",
-    "virtual-size": 1073741824,
-    "actual-size": 173000,
-    "backing-filename": "%s"
-}""" % (snap_file, src_volume['name'])
-        img_info = imageutils.QemuImgInfo(qemu_img_output, format='json')
+        qemu_img_output_snap = self.QEMU_IMG_INFO_TEMPLATE_SNAP % (
+            snap_file, src_volume['name'])
+        img_info_snap = imageutils.QemuImgInfo(qemu_img_output_snap,
+                                               format='json')
+
+        qemu_img_output_bck = self.QEMU_IMG_INFO_TEMPLATE_SRC % (
+            src_volume['name'])
+        img_info_src = imageutils.QemuImgInfo(
+            qemu_img_output_bck, format='json')
 
         # mocking and testing starts here
         mock_convert = self.mock_object(image_utils, 'convert_image')
         drv._read_info_file = mock.Mock(return_value=
                                         {'active': snap_file,
                                          snapshot['id']: snap_file})
-        self.mock_object(image_utils, 'qemu_img_info', return_value=img_info)
         drv._set_rw_permissions = mock.Mock()
         drv._create_overlay_volume_from_snapshot = mock.Mock()
+        shutil.copyfile = mock.Mock()
+        snap_info_call = mock.call(
+            snap_path,
+            force_share=True,
+            run_as_root=False,
+            allow_qcow2_backing_file=True)
+        src_info_call = mock.call(
+            src_vol_path,
+            force_share=True,
+            run_as_root=False,
+            allow_qcow2_backing_file=True)
+        self.mock_object(image_utils, 'qemu_img_info',
+                         side_effect=[img_info_snap, img_info_src])
 
         drv._copy_volume_from_snapshot(snapshot, dest_volume, size)
 
         drv._read_info_file.assert_called_once_with(info_path)
         os_ac_mock.assert_called_once_with(
             drv._local_volume_from_snap_cache_path(snapshot), os.F_OK)
-        image_utils.qemu_img_info.assert_called_once_with(
-            snap_path,
-            force_share=True,
-            run_as_root=False,
-            allow_qcow2_backing_file=True)
+        image_utils.qemu_img_info.assert_has_calls([snap_info_call,
+                                                    src_info_call])
         (mock_convert.
          assert_called_once_with(
              src_vol_path,
              drv._local_volume_from_snap_cache_path(snapshot), 'qcow2',
-             run_as_root=self._driver._execute_as_root))
+             run_as_root=self._driver._execute_as_root, data=img_info_src))
         os_sl_mock.assert_called_once_with(
             src_vol_path,
             drv._local_volume_from_snap_cache_path(snapshot) + '.parent-'
@@ -1180,37 +1224,46 @@ class QuobyteDriverTestCase(test.TestCase):
 
         size = dest_volume['size']
 
-        qemu_img_output = """{
-    "filename": "%s",
-    "format": "raw",
-    "virtual-size": 1073741824,
-    "actual-size": 173000,
-    "backing-filename": "%s"
-}""" % (snap_file, src_volume['name'])
-        img_info = imageutils.QemuImgInfo(qemu_img_output, format='json')
+        qemu_img_output_snap = self.QEMU_IMG_INFO_TEMPLATE_SNAP % (
+            snap_file, src_volume['name'])
+        img_info_snap = imageutils.QemuImgInfo(qemu_img_output_snap,
+                                               format='json')
+
+        qemu_img_output_bck = self.QEMU_IMG_INFO_TEMPLATE_SRC % (
+            src_volume['name'])
+        img_info_src = imageutils.QemuImgInfo(
+            qemu_img_output_bck, format='json')
 
         # mocking and testing starts here
         mock_convert = self.mock_object(image_utils, 'convert_image')
         drv._read_info_file = mock.Mock(return_value=
                                         {'active': snap_file,
                                          snapshot['id']: snap_file})
-        self.mock_object(image_utils, 'qemu_img_info', return_value=img_info)
+        snap_info_call = mock.call(
+            snap_path,
+            force_share=True,
+            run_as_root=False,
+            allow_qcow2_backing_file=True)
+        src_info_call = mock.call(
+            src_vol_path,
+            force_share=True,
+            run_as_root=False,
+            allow_qcow2_backing_file=True)
+        self.mock_object(image_utils, 'qemu_img_info',
+                         side_effect=[img_info_snap, img_info_src])
         drv._set_rw_permissions = mock.Mock()
         self.mock_object(shutil, 'copyfile')
 
         drv._copy_volume_from_snapshot(snapshot, dest_volume, size)
 
         drv._read_info_file.assert_called_once_with(info_path)
-        image_utils.qemu_img_info.assert_called_once_with(
-            snap_path,
-            force_share=True,
-            run_as_root=False,
-            allow_qcow2_backing_file=True)
+        image_utils.qemu_img_info.assert_has_calls([snap_info_call,
+                                                    src_info_call])
         (mock_convert.
          assert_called_once_with(
              src_vol_path,
              drv._local_volume_from_snap_cache_path(snapshot), 'raw',
-             run_as_root=self._driver._execute_as_root))
+             run_as_root=self._driver._execute_as_root, data=img_info_src))
         qb_falloc_mock.assert_called_once_with(dest_vol_path, size)
         shutil.copyfile.assert_called_once_with(cache_path, dest_vol_path)
         drv._set_rw_permissions.assert_called_once_with(dest_vol_path)
diff --git a/cinder/volume/drivers/quobyte.py b/cinder/volume/drivers/quobyte.py
index 86071c489..69f7afb5b 100644
--- a/cinder/volume/drivers/quobyte.py
+++ b/cinder/volume/drivers/quobyte.py
@@ -400,6 +400,8 @@ class QuobyteDriver(remotefs_drv.RemoteFSSnapDriverDistributed):
         # when this snapshot was created.
         img_info = self._qemu_img_info(forward_path, snapshot.volume.name)
         path_to_snap_img = os.path.join(vol_path, img_info.backing_file)
+        snap_backing_file_img_info = self._qemu_img_info(path_to_snap_img,
+                                                         snapshot.volume.name)
 
         path_to_new_vol = self._local_path_volume(volume)
         path_to_cached_vol = self._local_volume_from_snap_cache_path(snapshot)
@@ -416,7 +418,8 @@ class QuobyteDriver(remotefs_drv.RemoteFSSnapDriverDistributed):
             image_utils.convert_image(path_to_snap_img,
                                       path_to_new_vol,
                                       out_format,
-                                      run_as_root=self._execute_as_root)
+                                      run_as_root=self._execute_as_root,
+                                      data=snap_backing_file_img_info)
         else:
             # create the volume via volume cache
             if not os.access(path_to_cached_vol, os.F_OK):
@@ -425,7 +428,8 @@ class QuobyteDriver(remotefs_drv.RemoteFSSnapDriverDistributed):
                 image_utils.convert_image(path_to_snap_img,
                                           path_to_cached_vol,
                                           out_format,
-                                          run_as_root=self._execute_as_root)
+                                          run_as_root=self._execute_as_root,
+                                          data=snap_backing_file_img_info)
                 if self.configuration.quobyte_overlay_volumes:
                     # NOTE(kaisers): Create a parent symlink to track the
                     # existence of the parent
