diff --git a/cinder/image/image_utils.py b/cinder/image/image_utils.py
index 93e29dd5f..4e30d5f4e 100644
--- a/cinder/image/image_utils.py
+++ b/cinder/image/image_utils.py
@@ -387,7 +387,8 @@ def _convert_image(prefix: tuple,
                    passphrase_file: Optional[str] = None,
                    compress: bool = False,
                    src_passphrase_file: Optional[str] = None,
-                   disable_sparse: bool = False) -> None:
+                   disable_sparse: bool = False,
+                   allow_qcow2_backing_file: bool = False) -> None:
     """Convert image to other format.
 
     NOTE: If the qemu-img convert command fails and this function raises an
@@ -463,8 +464,11 @@ def _convert_image(prefix: tuple,
     if duration < 1:
         duration = 1
     try:
-        image_size = qemu_img_info(source,
-                                   run_as_root=run_as_root).virtual_size
+        image_size = qemu_img_info(
+            source,
+            run_as_root=run_as_root,
+            allow_qcow2_backing_file=allow_qcow2_backing_file
+        ).virtual_size
     except ValueError as e:
         msg = ("The image was successfully converted, but image size "
                "is unavailable. src %(src)s, dest %(dest)s. %(error)s")
@@ -499,7 +503,8 @@ def convert_image(source: str,
                   src_passphrase_file: Optional[str] = None,
                   image_id: Optional[str] = None,
                   data: Optional[imageutils.QemuImgInfo] = None,
-                  disable_sparse: bool = False) -> None:
+                  disable_sparse: bool = False,
+                  allow_qcow2_backing_file: bool = False) -> None:
     """Convert image to other format.
 
     NOTE: If the qemu-img convert command fails and this function raises an
@@ -524,7 +529,8 @@ def convert_image(source: str,
     :raises ImageUnacceptable: when the image fails some format checks
     :raises ProcessExecutionError: when something goes wrong during conversion
     """
-    check_image_format(source, src_format, image_id, data, run_as_root)
+    check_image_format(source, src_format, image_id, data, run_as_root,
+                       allow_qcow2_backing_file)
 
     if not throttle:
         throttle = throttling.Throttle.get_default()
@@ -539,7 +545,8 @@ def convert_image(source: str,
                        passphrase_file=passphrase_file,
                        compress=compress,
                        src_passphrase_file=src_passphrase_file,
-                       disable_sparse=disable_sparse)
+                       disable_sparse=disable_sparse,
+                       allow_qcow2_backing_file=allow_qcow2_backing_file)
 
 
 def resize_image(source: str,
@@ -791,7 +798,8 @@ def check_image_format(source: str,
                        src_format: Optional[str] = None,
                        image_id: Optional[str] = None,
                        data: Optional[imageutils.QemuImgInfo] = None,
-                       run_as_root: bool = True) -> None:
+                       run_as_root: bool = True,
+                       allow_qcow2_backing_file: bool = False) -> None:
     """Do some image format checks.
 
     Verifies that the src_format matches what qemu-img thinks the image
@@ -811,7 +819,8 @@ def check_image_format(source: str,
     if image_id is None:
         image_id = 'internal image'
     if data is None:
-        data = qemu_img_info(source, run_as_root=run_as_root)
+        data = qemu_img_info(source, run_as_root=run_as_root,
+                             allow_qcow2_backing_file=allow_qcow2_backing_file)
 
     if data.file_format is None:
         raise exception.ImageUnacceptable(
diff --git a/cinder/tests/unit/test_image_utils.py b/cinder/tests/unit/test_image_utils.py
index 4bd073c5d..fc1c9f7cd 100644
--- a/cinder/tests/unit/test_image_utils.py
+++ b/cinder/tests/unit/test_image_utils.py
@@ -315,7 +315,8 @@ class TestConvertImage(test.TestCase):
             output = image_utils.convert_image(source, dest, out_format,
                                                throttle=throttle)
 
-            my_call = mock.call(source, run_as_root=True)
+            my_call = mock.call(source, run_as_root=True,
+                                allow_qcow2_backing_file=False)
             mock_info.assert_has_calls([my_call, my_call])
             self.assertIsNone(output)
             mock_exec.assert_called_once_with('cgcmd', 'qemu-img', 'convert',
@@ -2662,7 +2663,8 @@ class TestImageFormatCheck(test.TestCase):
                                 image_utils.check_image_format,
                                 src)
         self.assertIn(expected_image_id, str(iue))
-        mock_info.assert_called_with(src, run_as_root=True)
+        mock_info.assert_called_with(src, run_as_root=True,
+                                     allow_qcow2_backing_file=False)
 
         # a VMDK should trigger an additional check
         mock_info.reset_mock()
@@ -2717,7 +2719,7 @@ class TestImageFormatCheck(test.TestCase):
         out_fmt = mock.sentinel.out_fmt
 
         image_utils.convert_image(src, dest, out_fmt)
-        mock_check.assert_called_once_with(src, None, None, None, True)
+        mock_check.assert_called_once_with(src, None, None, None, True, False)
 
 
 @ddt.ddt
diff --git a/cinder/tests/unit/volume/drivers/test_quobyte.py b/cinder/tests/unit/volume/drivers/test_quobyte.py
index fe422fc23..976747049 100644
--- a/cinder/tests/unit/volume/drivers/test_quobyte.py
+++ b/cinder/tests/unit/volume/drivers/test_quobyte.py
@@ -1019,7 +1019,8 @@ class QuobyteDriverTestCase(test.TestCase):
          assert_called_once_with(src_vol_path,
                                  dest_vol_path,
                                  'raw',
-                                 run_as_root=self._driver._execute_as_root))
+                                 run_as_root=self._driver._execute_as_root,
+                                 allow_qcow2_backing_file=True))
         drv._set_rw_permissions.assert_called_once_with(dest_vol_path)
 
     @mock.patch.object(quobyte.QuobyteDriver, "_fallocate_file")
@@ -1145,7 +1146,8 @@ class QuobyteDriverTestCase(test.TestCase):
          assert_called_once_with(
              src_vol_path,
              drv._local_volume_from_snap_cache_path(snapshot), 'qcow2',
-             run_as_root=self._driver._execute_as_root))
+             run_as_root=self._driver._execute_as_root,
+             allow_qcow2_backing_file=True))
         os_sl_mock.assert_called_once_with(
             src_vol_path,
             drv._local_volume_from_snap_cache_path(snapshot) + '.parent-'
@@ -1210,7 +1212,8 @@ class QuobyteDriverTestCase(test.TestCase):
          assert_called_once_with(
              src_vol_path,
              drv._local_volume_from_snap_cache_path(snapshot), 'raw',
-             run_as_root=self._driver._execute_as_root))
+             run_as_root=self._driver._execute_as_root,
+             allow_qcow2_backing_file=True))
         qb_falloc_mock.assert_called_once_with(dest_vol_path, size)
         shutil.copyfile.assert_called_once_with(cache_path, dest_vol_path)
         drv._set_rw_permissions.assert_called_once_with(dest_vol_path)
@@ -1388,7 +1391,8 @@ class QuobyteDriverTestCase(test.TestCase):
                 run_as_root=False,
                 allow_qcow2_backing_file=True)
             mock_convert_image.assert_called_once_with(
-                volume_path, upload_path, 'raw', run_as_root=False)
+                volume_path, upload_path, 'raw', run_as_root=False,
+                allow_qcow2_backing_file=True)
             mock_upload_volume.assert_called_once_with(
                 mock.ANY, mock.ANY, mock.ANY, upload_path, run_as_root=False,
                 store_id=None, base_image_ref=None, compress=True,
@@ -1447,7 +1451,8 @@ class QuobyteDriverTestCase(test.TestCase):
                 run_as_root=False,
                 allow_qcow2_backing_file=True)
             mock_convert_image.assert_called_once_with(
-                volume_path, upload_path, 'raw', run_as_root=False)
+                volume_path, upload_path, 'raw', run_as_root=False,
+                allow_qcow2_backing_file=True)
             mock_upload_volume.assert_called_once_with(
                 mock.ANY, mock.ANY, mock.ANY, upload_path, volume_fd=None,
                 run_as_root=False,
diff --git a/cinder/volume/drivers/quobyte.py b/cinder/volume/drivers/quobyte.py
index 86071c489..6bdea37ff 100644
--- a/cinder/volume/drivers/quobyte.py
+++ b/cinder/volume/drivers/quobyte.py
@@ -416,7 +416,8 @@ class QuobyteDriver(remotefs_drv.RemoteFSSnapDriverDistributed):
             image_utils.convert_image(path_to_snap_img,
                                       path_to_new_vol,
                                       out_format,
-                                      run_as_root=self._execute_as_root)
+                                      run_as_root=self._execute_as_root,
+                                      allow_qcow2_backing_file=True)
         else:
             # create the volume via volume cache
             if not os.access(path_to_cached_vol, os.F_OK):
@@ -425,7 +426,8 @@ class QuobyteDriver(remotefs_drv.RemoteFSSnapDriverDistributed):
                 image_utils.convert_image(path_to_snap_img,
                                           path_to_cached_vol,
                                           out_format,
-                                          run_as_root=self._execute_as_root)
+                                          run_as_root=self._execute_as_root,
+                                          allow_qcow2_backing_file=True)
                 if self.configuration.quobyte_overlay_volumes:
                     # NOTE(kaisers): Create a parent symlink to track the
                     # existence of the parent
diff --git a/cinder/volume/drivers/remotefs.py b/cinder/volume/drivers/remotefs.py
index 64bbeb225..f90f868e6 100644
--- a/cinder/volume/drivers/remotefs.py
+++ b/cinder/volume/drivers/remotefs.py
@@ -1103,7 +1103,8 @@ class RemoteFSSnapDriverBase(RemoteFSDriver):
                 # or volume data not being stored in raw format
                 #  (upload_volume assumes raw format input)
                 image_utils.convert_image(active_file_path, temp_path, 'raw',
-                                          run_as_root=self._execute_as_root)
+                                          run_as_root=self._execute_as_root,
+                                          allow_qcow2_backing_file=True)
                 upload_path = temp_path
             else:
                 upload_path = active_file_path
@@ -2266,7 +2267,8 @@ class RemoteFSManageableVolumesMixin(object):
     @typing.no_type_check
     def _get_rounded_manageable_image_size(self, image_path):
         image_size = image_utils.qemu_img_info(
-            image_path, run_as_root=self._execute_as_root).virtual_size
+            image_path, run_as_root=self._execute_as_root,
+            allow_qcow2_backing_file=True).virtual_size
         return int(math.ceil(float(image_size) / units.Gi))
 
     def manage_existing_get_size(self, volume, existing_ref):
