diff --git nova/virt/libvirt/imagebackend.py nova/virt/libvirt/imagebackend.py
index bc7f05c..737a1a9 100644
--- nova/virt/libvirt/imagebackend.py
+++ nova/virt/libvirt/imagebackend.py
@@ -246,9 +246,11 @@ class Image(object):
             if os.path.exists(base) and size > self.get_disk_size(base):
                 self.resize_image(size)
 
-            if (self.preallocate and self._can_fallocate() and
-                    os.access(self.path, os.W_OK)):
-                utils.execute('fallocate', '-n', '-l', size, self.path)
+            if (self.preallocate and os.access(self.path, os.W_OK)):
+                LOG.debug('Truncating new image at %(path)s to virtual size '
+                          '(patched for Quobyte).',
+                          {'path': self.path})
+                utils.execute('truncate', '-s', size, self.path)
 
     def _can_fallocate(self):
         """Check once per class, whether fallocate(1) is available,
