diff --git a/cinder/volume/drivers/quobyte.py b/cinder/volume/drivers/quobyte.py
index ede5e3f48..4720cc57f 100644
--- a/cinder/volume/drivers/quobyte.py
+++ b/cinder/volume/drivers/quobyte.py
@@ -71,7 +71,14 @@ volume_opts = [
                       ' quobyte_qcow2_volumes and'
                       ' quobyte_volume_from_snapshot_cache to be set to'
                       ' True. If one of these is set to False this option is'
-                      ' ignored.'))
+                      ' ignored.')),
+    cfg.BoolOpt('quobyte_multiattach',
+                default=False,
+                help=('Allows using Cinders multi attach feature with Quobyte.'
+                      ' Note that usage of multiattach with Quobyte requires'
+                      ' using an additional distributed locking service.'
+                      ' Ensure Quobyte volumes used with this option do not'
+                      ' use client caching in their volume configuration.'))
 ]

 CONF = cfg.CONF
@@ -700,6 +707,14 @@ class QuobyteDriver(remotefs_drv.RemoteFSSnapDriverDistributed):
             if self.configuration.quobyte_volume_from_snapshot_cache:
                 self._ensure_volume_from_snap_cache(mount_path)

+    def _update_volume_stats(self):
+        super(QuobyteDriver, self)._update_volume_stats()
+        if self.configuration.quobyte_multiattach:
+            self._stats["multiattach"] = True
+            LOG.info('multiattach for Quobyte is active.')
+        else:
+            self._stats["multiattach"] = False
+
     def _validate_volume(self, mount_path):
         """Runs a number of tests on the expect Quobyte mount"""
         partitions = psutil.disk_partitions(all=True)
